app/pages/lab-pages-editor/components/TasksPage/components/NewTaskButtonAndDialog.jsx

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable no-console */
+/* eslint-disable react/react-in-jsx-scope */
+/* eslint-disable react/require-default-props */
+/* eslint-disable radix */
+/* eslint-disable react/jsx-boolean-value */
+/* eslint-disable react/forbid-prop-types */
+/* eslint-disable react/no-unknown-property */
+
+import { useRef } from 'react';
+import PropTypes from 'prop-types';
+
+import CloseIcon from '../../../icons/CloseIcon.jsx';
+import TaskIcon from '../../../icons/TaskIcon.jsx';
+// import strings from '../../../strings.json'; // TODO: move all text into strings
+
+function NewTaskButtonAndDialog({
+  addTaskWithStep = () => {}
+}) {
+  const newTaskDialog = useRef(null);
+
+  function openNewTaskDialog() {
+    newTaskDialog.current?.showModal();
+  }
+
+  function closeNewTaskDialog() {
+    newTaskDialog.current?.close();
+  }
+
+  function addNewTask(e) {
+    const tasktype = e?.currentTarget?.dataset?.tasktype;
+    // Protip: don't use event.target, since it might return the child of the button, instead of the button itself
+
+    closeNewTaskDialog();
+    addTaskWithStep(tasktype);
+  }
+
+  return (
+    <>
+      <button
+        className="flex-item big primary"
+        onClick={openNewTaskDialog}
+        type="button"
+      >
+        Add a new Task +
```

Comment:
Pedantic point, but maybe add the '+' with `::after`, to give a more natural button label.